From c733d682e5762d61fb9df4439e95cc351fc80209 Mon Sep 17 00:00:00 2001
From: Sourav Poddar <sourav.poddar@ti.com>
Date: Mon, 25 Nov 2013 10:42:04 +0530
Subject: [PATCH 109/263] spi/qspi: cleanup memory mapped set up register.

This commit comes from branch ti-linux-3.12.y:
  git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git

Fix memory mapped set up register code to use
proper macros.

Signed-off-by: Sourav Poddar <sourav.poddar@ti.com>
(cherry picked from commit d475a8d223dd3dd626d744db38b8894e89bb3e19)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/spi/spi-ti-qspi.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/spi/spi-ti-qspi.c b/drivers/spi/spi-ti-qspi.c
index a7f378e..5b3083d 100644
--- a/drivers/spi/spi-ti-qspi.c
+++ b/drivers/spi/spi-ti-qspi.c
@@ -181,7 +181,7 @@ static int ti_qspi_setup(struct spi_device *spi)
 	struct ti_qspi	*qspi = spi_master_get_devdata(spi->master);
 	struct ti_qspi_regs *ctx_reg = &qspi->ctx_reg;
 	int clk_div = 0, ret;
-	u32 clk_ctrl_reg, clk_rate, clk_mask, memval = 0;
+	u32 clk_ctrl_reg, clk_rate, clk_mask, memval = 0, mode;
 	qspi->dc = 0;
 
 	if (spi->master->busy) {
@@ -240,13 +240,14 @@ static int ti_qspi_setup(struct spi_device *spi)
 	ti_qspi_write(qspi, qspi->dc, QSPI_SPI_DC_REG);
 
 	if (qspi->memory_mapped) {
-		switch (spi->mode) {
-		case SPI_TX_DUAL:
+		mode = spi->mode & (SPI_RX_DUAL | SPI_RX_QUAD);
+		switch (mode) {
+		case SPI_RX_DUAL:
 			memval |= (QSPI_CMD_DUAL_RD | QSPI_SETUP0_A_BYTES |
 				QSPI_SETUP0_8_BITS | QSPI_SETUP0_RD_DUAL |
 				QSPI_CMD_WRITE | QSPI_NUM_DUMMY_BITS);
 			break;
-		case SPI_TX_QUAD:
+		case SPI_RX_QUAD:
 			memval |= (QSPI_CMD_QUAD_RD | QSPI_SETUP0_A_BYTES |
 				QSPI_SETUP0_8_BITS | QSPI_SETUP0_RD_QUAD |
 				QSPI_CMD_WRITE | QSPI_NUM_DUMMY_BITS);
-- 
1.7.5.4

